kamon-newrelic/src/main/scala/kamon/newrelic/CustomMetricExtractor.scala

```diff
@@ -16,22 +16,31 @@
 
 package kamon.newrelic
 
-import kamon.metric.{ EntitySnapshot, Entity }
+import kamon.metric.{ MetricKey, EntitySnapshot, Entity }
 import kamon.metric.instrument.CollectionContext
 
 object CustomMetricExtractor extends MetricExtractor {
 
   def extract(settings: AgentSettings, collectionContext: CollectionContext, metrics: Map[Entity, EntitySnapshot]): Map[MetricID, MetricData] = {
-    def onlySimpleMetrics(kv: (Entity, EntitySnapshot)): Boolean =
-      kamon.metric.SingleInstrumentEntityRecorder.AllCategories.contains(kv._1.category)
+    val (simple, complex) = metrics.partition(simpleMetrics)
```

Comment:
Here the `metrics` map will contain all metrics, included trace and trace-segment metrics, they should be filtered out as they will already be processed by the web transaction metrics extractor.